enhancements/installer/byo-custom-dns.md

```diff
@@ -0,0 +1,573 @@
+public and private hosted zones.
+
+In addition, in OpenShift's Managed offering, the external DNS solution is also
+responsible for DNS records for required OpenShift resources including quay.io,
+registry.redhat.io, api.openshift.com, cloud management endpoints, etc. This
+enhancement proposal will not talk about how these entries are configured for
+the OpenShift Managed offering. 
+
+### User Stories
+
+"As a cloud admin, I want to run an OpenShift cluster on AWS with my custom DNS
+solution in place of the default cloud DNS."
+
+TODO: Is there anything else to add here?
+
+### Goals
+
+- Enable OpenShift customers to use their custom DNS solution instead of the
+default cloud solution.
+- Start with AWS and lay the groundwork to add this capability for other cloud
+platforms like Azure and GCP.
+- Continue using the default cloud LB service for API, API-Int and Ingress.
+
+### Non-Goals
+
+We will not use the [Kubernetes External DNS provider](https://github.com/openshift/external-dns)
+and/or the [OpenShift External DNS Operator](https://github.com/openshift/external-dns-operator)
+to program dns entries across varied DNS providers. The External DNS Operator
+relies on being able to program the customer's DNS solution. The customer DNS
+solutions in consideration here do not allow OpenShift to configure them.
+
+## Proposal
+
+For the OpenShift installation to make use of a custom DNS solution that is not
+configurable by OpenShift, the customer is responsible for configuring the DNS
+entries for the API, API-Int and Ingress services.
+
+Currently, during cluster installation, OpenShift is repsonsible for adding
+entries for API, API-Int and *.apps into the cloud DNS. These URLs are
+configured with the LB DNS Names of the LBs associated with each of these
+services. Hence the LBs have to be created before they can used to configure
+the DNS solution. Also, important to note is that these LB DNS Names cannot
+be generated before they are created.
+
+When it is indicated to the OpenShift installer via an input parameter that the
+custom DNS solution is in use, OpenShift does not configure the cloud DNS
+solution [Route53 for AWS](https://aws.amazon.com/route53/) for any of its
+services. In the case of API and API-Int services, the customer is responsible
+for creating the external and internal LBs [Elastic Load Balancer](https://aws.amazon.com/elasticloadbalancing/)
+prior to cluster installation. Their custom DNS solution also needs to be
+configured with entries for API and API-Int before cluster installation. OpenShift
+will continue to configure and manage these cloud LBs created by the user.
+
+DNS configuration for Ingress services will be handled differently. The Ingress
+Operator utilizes the upstream k8s LoadBalancer service to create and configure
+the LBs for the *.apps service. In this case, the customer pre-creating the LB
+would mean the LB service would have to be modified to consume a pre-created LB.
```

Comment:
Answered here: https://github.com/openshift/enhancements/pull/1400#discussion_r1273752479
What we lose is the ability to automatically add nodes to the LB when we scale out the cluster.

That's less important for the API LBs because we don't generally scale out the control planes. Although users using external DNS here will lose the benefits of MAPI-managed control planes when replacing a failed control plane node.